### PR TITLE
Character screen fixes

### PR DESCRIPTION
**Pull request by Vanshbajaj**
This PR introduces new features and UI updates to improve the user experience. The following changes have been made:

- Added Assisted Injection for Episode Screen.
- Updated the CharacterScreen layout and design to make it more intuitive.
- Fixed several UI inconsistencies and bugs reported by users.
- Optimized performance for faster load times and smoother interactions.

### Testing
- Verified that the new features are functional and meet the design requirements.

### CODE REVIEW

File: app/src/main/java/com/practical/assignment/AppNavGraph.kt [hunk 1]
```diff
@@ -25,7 +36,27 @@ fun AppNavGraph(navController: NavHostController, viewModelFactory: ViewModelFac
             val screen: CharacterScreenData = backStackEntry.toRoute()
             val characterViewModel: CharacterDetailsViewModel =
                 viewModel(factory = viewModelFactory)
-            CharacterScreen(screen.id, characterViewModel)
+            CharacterScreen(screen.id, characterViewModel) { episodeId ->
+                navController.navigate(EpisodeScreenData(episodeId))
```
Comment (anmolsahi-mm) [inline app/src/main/java/com/practical/assignment/AppNavGraph.kt:40]: generally, I prefer named arguments, but that's a personal preference and depends on the project coding conventions too. Not a big deal.

File: presentation/src/main/java/com/practical/presentation/screens/CharacterScreen.kt [hunk 2]
```diff
@@ -135,26 +153,82 @@ private fun TopData(character: CharacterModel, modifier: Modifier = Modifier) {
...
+
+@Composable
+private fun CharacterRow(
+    label: String,
+    value: String,
+    modifier: Modifier = Modifier,
+    color: Color = Purple40,
+) {
+    Row(
+        modifier = modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Start
```
Comment (anmolsahi-mm) [inline presentation/src/main/java/com/practical/presentation/screens/CharacterScreen.kt:168]: can we put the parameters on separate lines?

File: presentation/src/main/java/com/practical/presentation/screens/EspsoideDetailsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                when(state.exception){
+                    is NetworkException.ClientNetworkException->{
+
+                    }
+
+                }
+            }
+
+            is UiState.Success -> {
+
```
Comment (anmolsahi-mm) [inline presentation/src/main/java/com/practical/presentation/screens/EspsoideDetailsScreen.kt:37]: If this is a WIP, could you please add a TODO here?

Comment (sahianmol1) [reply presentation/src/main/java/com/practical/presentation/screens/EspsoideDetailsScreen.kt:37]: h

File: app/src/main/java/com/practical/assignment/MainActivity.kt [hunk 4]
```diff
@@ -7,13 +7,17 @@ import androidx.activity.enableEdgeToEdge
...
 import com.practical.presentation.ui.theme.PracticalAssignmentTheme
 import com.practical.presentation.factory.ViewModelFactory
+import dagger.assisted.AssistedFactory
 import javax.inject.Inject
 
 class MainActivity : ComponentActivity() {
     @Inject
     lateinit var viewModelFactory: ViewModelFactory
+    @Inject
+    lateinit var factory: EpisodeDetailsViewModelFactory
```
Comment (shreyansh-mm) [inline app/src/main/java/com/practical/assignment/MainActivity.kt:20]: I don't think we need 2 factories

File: presentation/src/main/java/com/practical/presentation/common/CommonComposable.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.practical.data.network.NetworkException
+import com.practical.presentation.R
+
+@Composable
+fun ErrorText(message: Int, modifier: Modifier = Modifier) {
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/practical/presentation/common/CommonComposable.kt:15]: private

File: presentation/src/main/java/com/practical/presentation/common/CommonComposable.kt
Comment (shreyansh-mm) [inline presentation/src/main/java/com/practical/presentation/common/CommonComposable.kt:1]: Should be named as `ErrorComponents` or `ErrorUi`

File: presentation/src/main/java/com/practical/presentation/di/EpisodeDetailsViewModelFactory.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            this.episodeId = episodeId
+        }
+
+        return episodeDetailsFactory.create(episodeId)
+    }
+
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(EpisodeDetailsViewModel::class.java)) {
+            // Ensure the episodeId is set before creating the ViewModel
+            checkNotNull(episodeId) { "Episode ID is missing" }
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/practical/presentation/di/EpisodeDetailsViewModelFactory.kt:26]: use requireNotNull

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

**Review by shreyansh-mm — APPROVED** [review]

**Pull request merged**